traits/trait_types.py

```diff
@@ -1835,25 +1835,28 @@ def validate(self, object, name, value):
     def float_validate(self, object, name, value):
         """ Validate that the value is a float value in the specified range.
         """
+        # Keep original value for use in error reporting.
+        original_value = value
         try:
-            if (
-                isinstance(value, RangeTypes)
-                and (
-                    (self._low is None)
-                    or (self._exclude_low and (self._low < value))
-                    or ((not self._exclude_low) and (self._low <= value))
-                )
-                and (
-                    (self._high is None)
-                    or (self._exclude_high and (self._high > value))
-                    or ((not self._exclude_high) and (self._high >= value))
-                )
-            ):
-                return float(value)
-        except:
-            pass
+            value = _validate_float(value)
+        except TypeError:
+            self.error(object, name, original_value)
 
-        self.error(object, name, value)
+        if (
```

Comment:
Note: content of this if block is unchanged, apart from the indentation. I'd like to rewrite this, but not in this PR.

Reply:
We're not performing the instance check in the if block anymore but that is sort of being handled by the use of `_validate_float`, right?

Reply:
Yes, sorry; that part of the content of the `if` block _is_ changed, and yes, the `_validate_float` check performs the equivalent type checking.